tests/api/printf.cpp

```diff
@@ -227,4 +234,35 @@ TEST_F(WithCommandQueue, PrintfMissingLengthModifier) {
     ASSERT_STREQ(printf_buffer, message);
 }
 
+void printf_callback(const char* buffer, size_t len, size_t complete,
+                     void* user_data) {
+    ASSERT_EQ(strlen(buffer), len);
+}
+
+TEST_F(WithPrintfEnabled, PrintSimple) {
+    const char message[] = "Hello World!";
+    char source[512];
```

Comment:
Maybe make a `#define` or something to have `"Hello World!" in a single spot and be able in `printf_callback` to compare the buffer to the expected string.